src/helper/crypt.rs

```diff
@@ -477,25 +477,29 @@ fn hash(algorithm: &str, buffers: Vec<&[u8]>) -> Result<Vec<u8>, String> {
     Ok(digest.finalize().to_vec())
 }
 
+#[inline]
 fn gen_random_16() -> Vec<u8> {
     let buf: &mut [u8] = &mut [0; 16];
     getrandom::getrandom(buf);
     buf.to_vec()
 }
 
+#[inline]
 fn gen_random_32() -> Vec<u8> {
     let buf: &mut [u8] = &mut [0; 32];
     getrandom::getrandom(buf);
     buf.to_vec()
 }
 
+#[inline]
 fn gen_random_64() -> Vec<u8> {
     let buf: &mut [u8] = &mut [0; 64];
     getrandom::getrandom(buf);
     buf.to_vec()
 }
 
 // Create a buffer of an integer encoded as a uint32le
+#[inline]
```

Comment:
Are we sure all of these are more suitable to be inline? Did we check what impact this has on binary size?

Reply:
Putting `#[inline]` on functions enable them to be optimized across crates.  This is a library crate, so if you don't have that and LTO is not on, the Rust compiler will never inline even very simple function such as getters.

Also, inlining usually enables more optimizations in the call site.  For example, the following code:
```rust
fn flip(x: bool) -> bool {
    if x { false } else { true }
}

// use it in another crate:
if flip(true) { ... }
```

If you don't compile with LTO, everything stays, including the call to `flip` because the Rust compiler will not inline across crate boundaries.  If you put `#[inline]` on `fn flip`, the entire code is optimized away to zero.

Reply:
There is a cost in terms of compile time by doing that optimization. All I was saying is that by doing this we remove that choice from the end user. Whereas with LTO they can opt in if they want to.

Reply:
AFAIK Rust doesn't inline at all in debug mode. That's why performance is way slower than release, because all those `as_ref` and `as_mut` etc are actually function calls where with inlining they turn to the original reference.

Reply:
> Fourth, when building libraries, proactively add #[inline] to small non-generic functions. Pay special attention to impls: Deref, AsRef and the like often benefit from inlining. A library can’t anticipate all usages upfront, it makes sense to not prematurely pessimize future users. Note that #[inline] is not transitive: if a trivial public function calls a trivial private function, you need to #[inline] both. See this benchmark for details.

This section is actually my exact point. A library should proactively `#[inline]` wherever it makes sense.